WcaOnRails/app/webpacker/components/CompetitionOverview/CompetitionFilters.js

```diff
@@ -0,0 +1,662 @@
+    setDelegatesInfo(flatData);
+
+    if (delegateHasNextPage) {
+      delegateFetchNextPage();
+    }
+  }, [delegatesData, delegateHasNextPage, delegateFetchNextPage]);
+
+  const customTimeSelectionButton = (
+    <Button
+      primary
+      name="state"
+      id="custom"
+      value="custom"
+      onClick={() => editTimeOrder('custom')}
+      active={timeOrder === 'custom'}
+    >
+      <span className="caption">{I18n.t('competitions.index.custom')}</span>
+    </Button>
+  );
+
+  return (
+    <Container>
+      <h2>{I18n.t('competitions.index.title')}</h2>
+      <Form className="competition-select" id="competition-query-form" acceptCharset="UTF-8">
+        <Form.Field>
+          <label htmlFor="events">
+            {`${I18n.t('competitions.competition_form.events')}`}
+            <br />
+            <Button primary type="button" size="mini" id="select-all-events" onClick={() => setSelectedEvents(WCA_EVENT_IDS)}>{I18n.t('competitions.index.all_events')}</Button>
+            <Button size="mini" id="clear-all-events" onClick={() => setSelectedEvents([])}>{I18n.t('competitions.index.clear')}</Button>
+          </label>
+
+          <div id="events">
+            {WCA_EVENT_IDS.map((eventId) => (
+              <React.Fragment key={eventId}>
+                <Button
+                  basic
+                  icon
+                  toggle
+                  size="mini"
+                  className="event-checkbox"
+                  id={`checkbox-${eventId}`}
+                  value={eventId}
+                  data-tooltip={I18n.t(`events.${eventId}`)}
+                  data-variation="tiny"
+                  onClick={() => editSelectedEvents(eventId)}
+                  active={selectedEvents.includes(eventId)}
+                >
+                  <Icon className={`cubing-icon event-${eventId}`} />
+                </Button>
+              </React.Fragment>
+            ))}
+          </div>
+        </Form.Field>
+
+        <Form.Group>
+          <Form.Field width={6}>
+            <label htmlFor="region">{I18n.t('competitions.index.region')}</label>
+            <Dropdown
+              search
+              selection
+              defaultValue="all"
+              options={regionsOptions}
+              onChange={(event, data) => editSelectedRegion(data.value)}
+            />
+          </Form.Field>
+          <Form.Field width={6}>
+            <label htmlFor="search">{I18n.t('competitions.index.search')}</label>
+            <Input
+              name="search"
+              id="search"
+              icon="search"
+              placeholder={I18n.t('competitions.index.tooltips.search')}
+              onChange={(event, data) => editSearchQuery(data.value)}
+            />
+          </Form.Field>
+        </Form.Group>
+
+        <Form.Group>
+          <Form.Field width={8}>
+            <label htmlFor="delegate">{I18n.t('layouts.navigation.delegate')}</label>
+            <Dropdown
```

Comment:
I was entirely unaware of the component `<WcaSearch />`, that would have been really helpful. But the dropdown does look clean and it enables search! Here's a demo. However loading up all the delegates in the initialization of the page costs a bit computationally. It might be bias but I like this style of dropdown better than the original search, please let me know what your thoughts are. Perhaps I should post this in Slack to gather more opinion. But if there's a clear advantage/purpose to sticking with `<WcaSearch />` please do not feel bad to turn this idea down.

https://github.com/thewca/worldcubeassociation.org/assets/58278719/c695aace-21d7-4881-bcfd-acb3d52c63e4

Reply:
I too like the Dropdown design more. But having two different components in the codebase doing the same thing doesn't look so good because it will be like reinventing the same wheel.

Is the UI design the only difference? If yes, is it possible to change the UI of WCA Search to the way you did in Dropdown? WcaSearch is also using Dropdown internally, so I guess implementing the design in WcaSearch would be possible. You can post this in Slack as well to gather more opinion.